### ISSUE TITLE
Plan of branch management

### ISSUE THREAD

**Issue by nao-pon** [Studio-42/elFinder#1117]
Toward the 2.0.0 and 2.1.0 release, I am considering the reconfiguration of the branch.

-  [x] 2.0 (Make from current 2.x) : for develop of 2.0
-  [x] 2.1 (As-is) : for develop of 2.1
-  [x] 2.x (Marge current 2.1) : for develop of main stream
-  [x] 2.0-src (New branch) : 2.0 build for release version
-  [x] 2.1-src (New branch) : 2.1 build for release version

**Comment by barryvdh** [comment]
Sorry for the late reply, but this has a few problems with Composer.

You tag, for example, 2.1.1, 2.1.2 etc. This can be required with:

```
composer require studio-42/elfinder:"2.1.x"
``` 

Which is great! But if you have your stability set to `dev`, you will get the version from the `2.1` branch, which doesn't include a build.

```
composer require studio-42/elfinder:"2.1.x@dev"
```

They probably should be reversed, eg. work on `2.1-src` and build to `2.1`. Or there should be an alias (https://getcomposer.org/doc/articles/aliases.md) but not 100% sure if that works correctly with those branch names.

The main benefit is that a package require elFinder, can just require `2.1.x` and don't care about the stability settings.

**Comment by nao-pon** [comment]
@barryvdh Thanks!

I was well understood. However, now I am turning the script, such as automatic build and packaging, I am worried that does not occur mistake in their modification.

` Composer require studio-42 / elfinder: "dev-2.1-src" ` In how about you?

**Comment by barryvdh** [comment]
Sorry, I don't really understand your question. Could you rephrase that?

`composer require studio-42/elfinder:"dev-2.1-src"` would probably work indeed, but for example, in laravel-elfinder, I [require `"studio-42/elfinder": "2.1.x",`](https://github.com/barryvdh/laravel-elfinder/blob/35f8485f87f3e5330762c25be1e4fbea7c92a476/composer.json#L15)

If people want the latest nightly, they could require  `studio-42/elfinder:"2.1.x@dev"` in their own composer file, or lower the stability to `dev` in the settings. It would still validate as `2.1.x`, so will work with my package. But if they need to require `dev-2.1-src`, it will lead to conflicts.

**Comment by nao-pon** [comment]
Sorry my poor English.

I'm afraid to make a mistake the correction of scripts, such as automated build etc. in my server. But this is my personal problem. It may be time-consuming, but will consider the modification.

**Comment by nao-pon** [comment]
I changed branch names.

- 2.x-src - Main development branch
- 2.1-src - 2.1 development branch
- 2.0-src - 2.0 development branch
- 2.1 - 2.1 nightly build branch
- 2.0 - 2.0 nightly build branch

**Comment by barryvdh** [comment]
Okay cool.

So what is the difference between them? You are still developing all three of them? Usually the 'active' development branch is master, but it doesn't really matter.

**Comment by nao-pon** [comment]
Okay I changed to

-  master - Main development branch
-  2.1-src - 2.1 development branch, auto build to 2.1 on commit
-  2.0-src - 2.0 development branch, auto build to 2.0 on commit
-  2.1 - 2.1 nightly build branch
-  2.0 - 2.0 nightly build branch
